### PR TITLE
Feature/basic pie

### PR DESCRIPTION
**Pull request by netashefer**

### CODE REVIEW

File: src/components/GraphContainer/GraphContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const GraphContainer = ({ spec, graphId, width, height }: GraphContainerProps) => {
+    const graph = useRecoilValue(graphsAtom)?.find(g => g.graphId === graphId);
+    const [aggregatedData, setData] = useState([]);
+    const ref = useRef<HighchartsReact.RefObject>();
+
+    useEffect(() => {
+        request();// eslint-disable-next-line
+    }, [graph?.dataSourceId]);
+
+    const request = async () => {
```
Comment (netashefer) [inline src/components/GraphContainer/GraphContainer.tsx:27]: neta-change it

File: src/components/Catalog/GraphItem/GraphItem.tsx [hunk 2]
```diff
@@ -37,7 +38,7 @@ const GraphItem = ({ graphId, title, type }: GraphItemProps) => {
             onMouseLeave={() => setIsHovered(false)}
             onDoubleClick={() => addWidget(graphId)}
         >
-            <Icon className="graph-icon" />
+            <GraphTypeIcon className="graph-icon" />
             <div className='title'>{title}</div>
             {isHovered && <DeleteIcon className='delete-icon' onClick={deleteGraph} />}
         </div>
```
Comment (YahelR) [inline src/components/Catalog/GraphItem/GraphItem.tsx:44]: This comment is not related to this file, I just can't comment on the other one -
there is an empty file called GraphContainer.scss

File: src/components/GraphContainer/GraphContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+const GraphContainer = ({ graph, width, height }: GraphContainerProps) => {
+    const [aggregatedData, setData] = useState([]);
+    const ref = useRef<HighchartsReact.RefObject>();
+
+    useEffect(() => {
+        askAggregatedData();// eslint-disable-next-line
+    }, [graph?.dataSourceId]);
+
+    const askAggregatedData = async () => {
```
Comment (YahelR) [inline src/components/GraphContainer/GraphContainer.tsx]: change ask to 'get' or 'fetch'

File: src/components/GraphContainer/GraphContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    };
+
+    useEffect(() => {
+        ref?.current.chart.reflow();
+    }, [width, height]);
+
+    const options: Partial<Highcharts.Options> = {
+        ...GraphOptionsMap[graph?.template?.type],
+        chart: {
+            backgroundColor: "#2D325A",
```
Comment (YahelR) [inline src/components/GraphContainer/GraphContainer.tsx]: can u move this to a constant and move the constant to another file?

File: src/components/WidgetContainer/WidgetContainer.tsx [hunk 5]
```diff
@@ -1,29 +1,55 @@
...
 export const WIDGET_DRAGGABLE_TITLE_CLASSNAME = "widget-draggable-title";
 
 const WidgetContainer = ({ widgetId, graphId, widgetProps }: WidgetContainerProps) => {
     const removeWidget = useRemoveWidget();
+    const containerRef = useRef();
+    const [width, setWidth] = useState(0);
+    const [height, setHeight] = useState(0);
+    const graph = useRecoilValue(graphItemSelector(graphId));
+
+    useEffect(() => {
```
Comment (YahelR) [inline src/components/WidgetContainer/WidgetContainer.tsx:23]: is this useEffect supposed to have no dependencies?

File: src/components/WidgetContainer/WidgetContainer.tsx [hunk 6]
```diff
@@ -1,29 +1,55 @@
...
         <div
             className="widget-container"
             key={widgetId}
+            ref={containerRef}
         >
             <div className='widget-top'>
                 <EditIcon className='edit-icon' />
                 <p className={WIDGET_DRAGGABLE_TITLE_CLASSNAME}>
-                    {widgetId}
+                    {graph?.title || "no title"}
```
Comment (YahelR) [inline src/components/WidgetContainer/WidgetContainer.tsx]: "No Title"

File: src/helpers/graph.option.helper.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+                cursor: 'pointer',
+                dataLabels: {
+                    enabled: true,
+                    format: '{point.name}: {point.y}'
+                },
+                showInLegend: true
+            }
+        },
+    },
+
```
Comment (YahelR) [inline src/helpers/graph.option.helper.ts]: unnecessary whiteline

File: src/recoil/graphs/graphs.ts [hunk 8]
```diff
@@ -20,3 +20,10 @@ export const graphsAtom = atom<Graph[]>({
     key: 'graphsAtomKey',
     default: graphsDefaultSelector,
 });
+
+export const graphItemSelector = selectorFamily<Graph, string>({
```
Comment (YahelR) [inline src/recoil/graphs/graphs.ts]: can't we just call it graphSelector?

item is unnecessary to the name in my opinion

**Review by YahelR — CHANGES_REQUESTED** [review]
